### PR TITLE
zip support for URL provider

### PR DESCRIPTION
**Pull request by simonvh**
* Download zipped genomes and annotation
* Changed all `os.rename` to `shutil.move` to enable cross-device renaming on the way.

### CODE REVIEW

**Review by siebrenf — APPROVED** [review]
Nice! Didn't even know about shutil.move being more portable!

If tarring is only used on genomes, this is perfect.
If it is also be done with gene annotations, we could support that too without too much hassle.

**Comment by simonvh** [comment]
> Nice! Didn't even know about shutil.move being more portable!

I didn't either  :-D. I had to figure this out because tests were failing on my system.

> If tarring is only used on genomes, this is perfect.
> If it is also be done with gene annotations, we could support that too without too much hassle.

Maybe with a boolean argument `concat` to decide behaviour on multiple files, `concat=True` for genomes and `concat=False` for annotations (which should nog be combined in a bigfile)?

**Comment by simonvh** [comment]
Update! See what you think. I've also added a context manager:

```
with extracted_file(fname) as unzipped_fname:
    # do what you want with extracted file, and context manager makes sure it's gzipped afterwards
```

File: genomepy/files.py [hunk 1]
```diff
@@ -122,21 +146,126 @@ def gunzip_and_name(fname: str) -> Tuple[str, bool]:
...
+    -------
+    tuple
+        fname, str
+            up to date filename
+    """
+    if not outfile:
+        outfile = fname[:-4]
+
+    with ZipFile(fname, "r") as fzip:
+        with TemporaryDirectory() as tmpdir:
```
Comment (siebrenf) [inline genomepy/files.py:252]: `with ZipFile(fname, "r") as fzip, TemporaryDirectory() as tmpdir:`

just cleans up the indenting a bit

**Review by siebrenf — CHANGES_REQUESTED** [review]
1 bug in `extract_archive`, and `get_file_info()` cant recognize tar files yet. Rest is all tips!

File: genomepy/files.py [hunk 2]
```diff
@@ -122,21 +146,126 @@ def gunzip_and_name(fname: str) -> Tuple[str, bool]:
...
-            whether the file was gunzipped
     """
+    if fname.endswith(".gz"):
+        return extract_gzip(fname, outfile=outfile)
+    elif fname.endswith(
+        ".zip",
+    ):
+        return extract_zip(fname, outfile=outfile, concat=concat)
+    elif fname.endswith(".tgz") or fname.endswith(".tar.gz"):
+        return extract_tarball(fname, outfile=outfile)
```
[the remark this answers is not in the document]
Comment (siebrenf) [reply genomepy/files.py:168]: also, endswith accepts tuples! So you can reduce this to `.endswith((".tgz", ".tar.gz"))`

File: genomepy/files.py [hunk 3]
```diff
@@ -122,21 +146,126 @@ def gunzip_and_name(fname: str) -> Tuple[str, bool]:
...
+
```
[the remark this answers is not in the document]
Comment (simonvh) [reply genomepy/files.py:169]: It now returns `None` if extraction fails, which can be checked.

**Review by siebrenf — APPROVED** [review]

**Pull request merged**